contracts/rbn-staking/Redeemer.sol

```diff
@@ -0,0 +1,95 @@
+// SPDX-License-Identifier: AGPL-3.0-or-later
+pragma solidity ^0.8.0;
+
+import {ISeizer} from "../interfaces/ISeizer.sol";
+import {IVotingEscrow} from "../interfaces/IVotingEscrow.sol";
+import {IERC20} from "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import {
+  SafeERC20
+} from "@openzeppelin/contracts/token/ERC20/utils/SafeERC20.sol";
+import "@openzeppelin/contracts/utils/math/SafeMath.sol";
+
+contract Redeemer {
+  using SafeERC20 for IERC20;
+  using SafeMath for uint256;
+
+  // Multisig
+  address public admin;
+  // Maximum % of rbn staked redeemable. 2 decimals i.e 100 * 10 ** 2 = 100% possible to redeem
+  uint8 public maxRedeemPCT;
+  // Seizer implementation
+  ISeizer public seizerImplementation;
+  // veCRV escrow contract
+  IVotingEscrow public votingEscrowContract;
+
+  event RBNRedeemed(uint256 amountRedeemed);
+
+  constructor(address _admin, uint8 _maxRedeemPCT) {
+    require(_admin != address(0), "!_admin");
+    require(_maxRedeemPCT > 0 && _maxRedeemPCT < 10000, "!_maxRedeemPCT");
+
+    admin = _admin;
+    maxRedeemPCT = _maxRedeemPCT;
+  }
+
+  /**
+   * @dev Validates that the tx sender is admin multisig
+   */
+  modifier onlyAdmin() {
+    require(msg.sender == admin, "Must be admin");
+    _;
+  }
+
+  /**
+   * @dev Set new voting escrow contract
+   * @param _votingEscrowContract new voting escrow contract
+   */
+  function setVotingEscrowContract(address _votingEscrowContract)
+    external
+    onlyAdmin
+  {
+    require(_votingEscrowContract != address(0), "!_votingEscrowContract");
+    votingEscrowContract = IVotingEscrow(_votingEscrowContract);
+  }
+
+  /**
+   * @dev Set new seizer contract implementation
+   * @param _seizerImplementation new seizer contract
+   */
+  function setSeizerImplementation(address _seizerImplementation)
+    external
+    onlyAdmin
+  {
+    seizerImplementation = ISeizer(_seizerImplementation);
+  }
+
+  /**
+   * @dev Set new max redeemeable pct
+   * @param _maxRedeemPCT new max redeem pct
+   */
+  function setMaxRedeemPCT(uint8 _maxRedeemPCT) external onlyAdmin {
+    require(_maxRedeemPCT > 0 && _maxRedeemPCT < 10000, "!_maxRedeemPCT");
+    maxRedeemPCT = _maxRedeemPCT;
+  }
+
+  /**
+   * @dev Redeems the rbn
+   * @param _maxRedeemPCT new max redeem pct
+   */
+  function redeemRBN(uint256 _amount) external onlyAdmin {
+    require(votingEscrowContract != address(0), "!votingEscrowContract");
+
+    uint256 amountToRedeem =
+      seizerImplementation == address(0)
+        ? _amount
+        : seizerImplementation.amountToRedeem(votingEscrowContract);
```

Comment:
Following [this idea](https://github.com/ribbon-finance/governance/pull/40/files#r746418072), we can add the logic for determining the amount to redeem on this contract itself

Reply:
see above. the amount to redeem will depend on the rules we set up later which should be abstracted away to the implementation

 1. rule 1: x% threshold of vault x being ITM we sell that much rbn and deposit into that vault
 2. rule 2: smart contract (this should not be hardcoded) it will be subjective the amount etc.